packages/node_modules/@node-red/editor-client/src/js/ui/palette-editor.js

```diff
@@ -31,17 +31,75 @@ RED.palette.editor = (function() {
     var eventTimers = {};
     var activeFilter = "";
 
-    function semVerCompare(A,B) {
-        var aParts = A.split(".").map(function(m) { return parseInt(m);});
-        var bParts = B.split(".").map(function(m) { return parseInt(m);});
-        for (var i=0;i<3;i++) {
-            var j = aParts[i]-bParts[i];
-            if (j<0) { return -1 }
-            if (j>0) { return 1 }
+    var semverre = /^(?<major>\d+)(\.(?<minor>\d+))?(\.(?<patch>\d+))?(-(?<pre>[0-9A-Za-z-]+))?(\.(?<build>[0-9A-Za-z-.]+))?$/;
+	var NUMBERS_ONLY = /^\d+$/;
+
+
+	function SemVerPart ( part ) {
+        this.number = 0;
+        this.text = part;
+        if ( isNumeric( part ) )
+        {
+            this.number = parseInt( part );
+            this.type = "N";
+        } else
+        {
+            this.type = part == undefined || part.length < 1 ? "E" : "T";
         }
-        return 0;
     }
 
+    SemVerPart.prototype.compare = function ( other ) {
+        const types = this.type + other.type;
+
+        switch ( types )
+        {
+            case "EE": return 0;
+
+            case "NT":
+            case "TE":
+            case "EN": return -1;
+
+            case "NN": return this.number - other.number;
+
+            case "TT": return this.text.localeCompare( other.text );
+
+            case "ET":
+            case "TN":
+            case "NE": return 1;
+        }
+    };
+
+    function SemVer ( ver ) {
+        const { major, minor, patch, pre, build } = ver.match( semverre ).groups;
+        this.parts = [ new SemVerPart( major ), new SemVerPart( minor ), new SemVerPart( patch ), new SemVerPart( pre ), new SemVerPart( build ) ];
```

Comment:
Not sure if IE11 (which we try to keep working) will like this syntax.

```suggestion
        const groups = ver.match( semverre ).groups;
        this.parts = [ new SemVerPart( groups.major ), new SemVerPart( groups.minor ), new SemVerPart( groups.patch ), new SemVerPart( groups.pre ), new SemVerPart( groups.build ) ];
```